components/scroller/scroller.vue

```diff
@@ -0,0 +1,299 @@
+<template>
+  <core-scroller
+    ref="scroller"
+    :items="itemsWithSize"
+    :min-item-size="minItemSize"
+    :direction="direction"
+    key-field="id"
+    :list-tag="listTag"
+    :item-tag="itemTag"
+    v-bind="$attrs"
+  >
+    <template
+      v-if="!dynamic"
+      #default="{ item: itemWithSize, index, active }"
+    >
+      <slot
+        v-bind="{
+          item: itemWithSize.item,
+          index,
+          active,
+          itemWithSize,
+        }"
+      />
+    </template>
+
+    <template
+      v-else
+      #default="{ item: itemWithSize, index, active }"
+    >
+      <dt-scroller-item
+        :item="itemWithSize"
+        :active="active"
+        :size-dependencies="[
+          itemWithSize.message,
+        ]"
+        :data-index="index"
+      >
+        <slot
+          v-bind="{
+            item: itemWithSize.item,
+            index,
+            active,
+            itemWithSize,
+          }"
+        />
+      </dt-scroller-item>
+    </template>
+  </core-scroller>
+</template>
+
+<script>
+import CoreScroller from './modules/core-scroller.vue';
+import DtScrollerItem from './modules/scroller-item.vue';
+
+export default {
+  name: 'DynamicScroller',
+
+  components: {
+    CoreScroller,
+    DtScrollerItem,
+  },
+
+  provide () {
+    if (typeof ResizeObserver !== 'undefined') {
+      this.$_resizeObserver = new ResizeObserver(entries => {
+        requestAnimationFrame(() => {
+          if (!Array.isArray(entries)) {
+            return;
+          }
+          for (const entry of entries) {
+            if (entry.target && entry.target.$_vs_onResize) {
+              let width, height;
+              if (entry.borderBoxSize) {
+                const resizeObserverSize = entry.borderBoxSize[0];
+                width = resizeObserverSize.inlineSize;
+                height = resizeObserverSize.blockSize;
+              } else {
+                // @TODO remove when contentRect is deprecated
+                width = entry.contentRect.width;
+                height = entry.contentRect.height;
+              }
+              entry.target.$_vs_onResize(entry.target.$_vs_id, width, height);
+            }
+          }
+        });
+      });
+    }
```

Comment:
Also seems like library code? Just saying that due to the references in it so correct me if I'm wrong.

Reply:
Yes this is library code, Im still trying to trigger it to see if it necessary

Reply:
I'm not sure we should just be removing things if we don't know what they do, but it should be in the same file as the rest of the library code.

Reply:
I would not like to have unnecessary or "magic" code in our component, if it is not necessary I would remove it.

If in the further feedback we found a bug or miss some functionality we can work it later.

Reply:
Yep no problem, just making sure we know exactly what it does before removing.